### ISSUE TITLE
Local Changes refresh fails 

### ISSUE THREAD

**Issue by alexcomsa** [groboclown/p4ic4idea#206]
**Describe the bug**
I cannot read / refresh the  existing change lists from the Versions Control / Local Changes tab.
The plugin fails with the following message P4 Plugin Error: Internal error: null

**To Reproduce**
1. Create a new Change List in p4 client  
2. Set the CL as active in the IDE  
3. Submit the Change List to the stream with p4 client 
4. Try to refresh the VCS changes 
5. See error

**Expected behavior**
All existing CL's are displayed 

**Attachments**

```
019-11-11 11:12:55,297 [2629146]  ERROR - i.vcs.changes.ChangeListWorker - IntelliJ IDEA 2019.2.3  Build #IC-192.6817.14 
2019-11-11 11:12:55,297 [2629146]  ERROR - i.vcs.changes.ChangeListWorker - JDK: 11.0.4; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o 
2019-11-11 11:12:55,297 [2629146]  ERROR - i.vcs.changes.ChangeListWorker - OS: Windows 10 
2019-11-11 11:12:55,297 [2629146]  ERROR - i.vcs.changes.ChangeListWorker - Plugin to blame: Perforce IDEA Community Integration version: 0.10.8 
2019-11-11 11:12:55,297 [2629146]  ERROR - i.vcs.changes.ChangeListWorker - Changelist not found: vcs - p4ic 
java.lang.Throwable: Changelist not found: vcs - p4ic
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:145)
	at com.intellij.openapi.vcs.changes.ChangeListWorker$ChangeListUpdater.addChangeToList(ChangeListWorker.java:1098)
	at com.intellij.openapi.vcs.changes.UpdatingChangeListBuilder.processChangeInList(UpdatingChangeListBuilder.java:73)
	at net.groboclown.p4plugin.extension.P4ChangeProvider.associateChangeWithList(P4ChangeProvider.java:622)
	at net.groboclown.p4plugin.extension.P4ChangeProvider.processModifiedFileChange(P4ChangeProvider.java:516)
	at net.groboclown.p4plugin.extension.P4ChangeProvider.lambda$updateLocalFileCache$2(P4ChangeProvider.java:475)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at net.groboclown.p4plugin.extension.P4ChangeProvider.updateLocalFileCache(P4ChangeProvider.java:465)
	at net.groboclown.p4plugin.extension.P4ChangeProvider.updateFileCache(P4ChangeProvider.java:417)
	at net.groboclown.p4plugin.extension.P4ChangeProvider.getChanges(P4ChangeProvider.java:236)
	at com.intellij.openapi.vcs.changes.ChangeListManagerImpl.actualUpdate(ChangeListManagerImpl.java:716)
	at com.intellij.openapi.vcs.changes.ChangeListManagerImpl.iterateScopes(ChangeListManagerImpl.java:629)
	at com.intellij.openapi.vcs.changes.ChangeListManagerImpl.lambda$updateImmediately$13(ChangeListManagerImpl.java:536)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.vcs.changes.ChangeListManagerImpl.updateImmediately(ChangeListManagerImpl.java:497)
	at com.intellij.openapi.vcs.changes.ChangeListManagerImpl.lambda$new$0(ChangeListManagerImpl.java:145)
	at com.intellij.openapi.vcs.changes.UpdateRequestsQueue$MyRunnable.run(UpdateRequestsQueue.java:226)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:224)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

```
**Desktop (please complete the following information):**
 - OS: Windows10
 - IDE and version :IntelliJ IDEA 2019.2.3  Build #IC-192.6817.14
 - Version 0.10.9

**Comment by groboclown** [comment]
Thanks for reporting this issue.  I'm looking into it.

**Comment by groboclown** [comment]
For my reference:

The recent versions (>= v191 at least) have changed around the `ChangeListWorker` to now store changelists internally by name.  The plugin searches the `ChangelistManager` to ensure the changelist is registered before using it, but there's a disconnect somewhere between the two.  I'm investigating what's happening here to cause this scenario.

**Comment by groboclown** [comment]
@alexcomsa are you seeing this issue happen often, or was this a one-off thing?  That may help me narrow down whether this is a race condition or if something changed in the IDE.

**Comment by groboclown** [comment]
I see now.  This is the committed changelist bug.  This is a particularly nasty piece of work.

**Comment by alexcomsa** [comment]
@groboclown It happend after I submitted the pendig CL .
After several minutes the plugin recoverd . I'm not sure how it happend 
* I invalidated the cache 
* Disabled / Enabled the plugin
* Reimported the project in the IDE

**Comment by groboclown** [comment]
I have a fix coming out in v0.10.10
